branch-protection/src/index.ts

```diff
@@ -0,0 +1,77 @@
+import {getInput, info, warning, isDebug, debug, error} from "@actions/core"
+import {context as GHContext, getOctokit} from "@actions/github"
+
+async function run() {
+  const args = getArgs()
+  const octokit = getOctokit(args.gh_token)
+
+  let protection:any = await octokit.request("GET /repos/{owner}/{repo}/branches/{branch}/protection", {
+    owner: args.owner,
+    repo: args.repo,
+    branch: args.branch
+  })
+  debugOutput("protection", protection)
+  let success = false
+  if (protection.data.enforce_admins.enabled !== args.enforce_admins) {
+    protection.data.enforce_admins.enabled = args.enforce_admins
+  for (let i = 0; i <= args.retries; i++) {
+    try {
+
+        await octokit.request("PUT /repos/{owner}/{repo}/branches/{branch}/protection", protection.data )
+        warning(`Change enforce_admins state to ${args.enforce_admins.toString()}.`)
+
+      success = true
+      break
+    } catch (e) {
+      warning(`Failed to set enforce_admins to ${args.enforce_admins.toString()}. Retrying...`)
+      warning(e)
+      success = false
+      await sleep((i * 1000) ** 2)
+    }
+  }
```

Comment:
The indentation does not fit here. Would be best to use https://prettier.io/ for formatting.

Reply:
run reformat: done